src/Core/tree_policies/UCT_TreePolicy.cpp

```diff
@@ -0,0 +1,17 @@
+#include <UCT_TreePolicy.h>
+
+SearchNode UCT_TreePolicy::treePolicy(SearchNode &node) {
+
+    SearchNode c_node = node;
```

Comment:
This assignment implicitly calls the copy constructor of the node. Use SearchNode &c_node to avoid this